sklearn/cluster/tests/test_bicluster.py

```diff
@@ -231,33 +232,25 @@ def test_perfect_checkerboard():
                            (rows, cols)) == 1
 
 
-def test_errors():
+@pytest.mark.parametrize(
+    "args",
+    [{'n_clusters': (3, 3, 3)}, {'n_clusters': 'abc'},
+     {'n_clusters': (3, 'abc')}, {'method': 'unknown'},
+     {'n_components': 0}, {'n_best': 0},
+     {'svd_method': 'unknown'}]
+)
+def test_errors(args):
     data = np.arange(25).reshape((5, 5))
 
-    model = SpectralBiclustering(n_clusters=(3, 3, 3))
-    assert_raises(ValueError, model.fit, data)
-
-    model = SpectralBiclustering(n_clusters='abc')
-    assert_raises(ValueError, model.fit, data)
-
-    model = SpectralBiclustering(n_clusters=(3, 'abc'))
-    assert_raises(ValueError, model.fit, data)
-
-    model = SpectralBiclustering(method='unknown')
-    assert_raises(ValueError, model.fit, data)
-
-    model = SpectralBiclustering(svd_method='unknown')
-    assert_raises(ValueError, model.fit, data)
-
-    model = SpectralBiclustering(n_components=0)
-    assert_raises(ValueError, model.fit, data)
-
-    model = SpectralBiclustering(n_best=0)
-    assert_raises(ValueError, model.fit, data)
+    model = SpectralBiclustering(**args)
+    with pytest.raises(ValueError):
+        model.fit(data)
 
     model = SpectralBiclustering(n_components=3, n_best=4)
-    assert_raises(ValueError, model.fit, data)
+    with pytest.raises(ValueError):
+        model.fit(data)
 
     model = SpectralBiclustering()
     data = np.arange(27).reshape((3, 3, 3))
-    assert_raises(ValueError, model.fit, data)
+    with pytest.raises(ValueError):
```

Comment:
To avoid runnign this one for every parameter I guess you can just create a new test just for this and call it `test_wrong_shape`

Reply:
I will make this change